code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -884,6 +887,86 @@
 		new /obj/item/stack/cable_coil(loc, 3)
 	qdel(src)
 
+/obj/item/circuit_component/air_alarm
+	display_name = "Air Alarm"
+	display_desc = "Controls levels of gases and their temperature as well as all vents and scrubbers in the room."
+
+	var/datum/port/input/min_2
+	var/datum/port/input/min_1
+	var/datum/port/input/max_1
+	var/datum/port/input/max_2
+
+	var/datum/port/input/request_data
+
+	var/datum/port/output/pressure
+	var/datum/port/output/temperature
+	var/datum/port/output/gas_amount
+
+	var/obj/machinery/airalarm/connected_alarm
+	var/list/options_map
+
+/obj/item/circuit_component/air_alarm/Initialize()
+	. = ..()
+	min_2 = add_input_port("Min 2", PORT_TYPE_NUMBER)
+	min_1 = add_input_port("Min 1", PORT_TYPE_NUMBER)
+	max_1 = add_input_port("Max 1", PORT_TYPE_NUMBER)
+	max_2 = add_input_port("Max 2", PORT_TYPE_NUMBER)
+	request_data = add_input_port("Request Atmosphere Data", PORT_TYPE_SIGNAL)
+
+	pressure = add_output_port("Pressure", PORT_TYPE_NUMBER)
+	temperature = add_output_port("Temperature", PORT_TYPE_NUMBER)
+	gas_amount = add_output_port("Chosen Gas Amount", PORT_TYPE_NUMBER)
+
+/obj/item/circuit_component/air_alarm/populate_options()
+	var/static/list/component_options = list(
+		"Pressure",
+		"Temperature"
+	)
+	options = component_options
+
+	var/static/list/options_to_key = list(
+		"Pressure" = "pressure",
+		"Temperature" = "temperature"
+	)
+
+	for(var/gas_type in subtypesof(/datum/gas))
+		var/datum/gas/new_gas = new gas_type()
+		component_options.Add(new_gas.name)
```

Comment:
Because component_options is static, and this is on populate_options, would seem to me this would constantly fill this list with more and more.

Make it so this only runs once.

Reply:
Also don't instantiate them. Just use `initial(gas_type.name)`, after doing `for (var/datum/gas/gas_type as anything in subtypesof(/datum/gas))`